backdoor: allow avoiding iopl permission errors

lucab opened the pull request:
This introduces additional methods for the backdoor-opening logic,
in order to allow consuming applications to keep going when changing
I/O access level is not allowed (see kernel_lockdown(7) on Linux).
In most cases the library may work without higher I/O level,
and actual failures can be handled via normal error-handling on RPC
methods.

Ref: https://github.com/lucab/vmw_backdoor-rs/issues/6

cgwalters commented on src/backdoor.rs:
```diff
@@ -18,20 +18,20 @@ pub(crate) const COMMAND_ERPC: u32 = 0x1E;
 /// On Linux, this requires running with `CAP_SYS_RAWIO` privileges.
-pub fn probe_backdoor() -> Result<BackdoorGuard, VmwError> {
-    BackdoorGuard::change_io_access(true)?;
+pub fn probe_backdoor(ignore_access_errors: bool) -> Result<BackdoorGuard, VmwError> {
```
I would say a separate `probe_backdoor_optional` method is cleaner than passing a `bool`.  Take for example [open_file_optional](https://github.com/cgwalters/openat-ext/blob/64ca702d60308aabbd220a724f6f789ef6d99606/src/lib.rs#L31) vs `open_file(true)` or so.

Boolean arguments are just harder to read because you need to go to the call site to understand what they're doing.  And it's easy to get them backwards; is the boolean saying to ignore errors or to return them, etc.

cgwalters replied:
(Now internally in the code they can often end up calling a private helper that takes a bool, but the goal is to avoid confusion for people calling the public methods)

lucab replied:
Ah, good suggestion, the `_optional` methods would fit nice here. I'll rework in that direction.

Commit:
backdoor: allow avoiding iopl permission errors

This introduces additional methods for the backdoor-opening logic,
in order to allow consuming applications to keep going when changing
I/O access level is not alloewd (see kernel_lockdown(7) on Linux).
In most cases the library may work without higher I/O level,
and actual failures can be handled via normal error-handling on RPC
methods.

lucab commented:
I've repurposed the existing methods and added a new pair of `_privileged` methods on the side, how does that look to you?

I was actually unsure whether to do the opposite, i.e. keeping the existing methods with the `iopl` and adding `_unprivileged` ones without on the side. Thoughts?

cgwalters reviewed (approved):
I didn't deep dive into the logic, but superficially LGTM

lucab commented:
@cgwalters thanks for the feedback!
